sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -1355,9 +1377,10 @@ public void populateDisplayData(DisplayData.Builder builder) {
      * <p>Public so can be suppressed by runners.
      */
     public class PubsubBoundedWriter extends DoFn<PubsubMessage, Void> {
-      private transient List<OutgoingMessage> output;
+      private transient Map<PubsubTopic, List<OutgoingMessage>> output;
```

Comment:
could have a single map by topic, reduces lookup and prevents possibility of entry in one map and not the other.

Reply:
done